git-clone/README.md

```diff
@@ -50,3 +50,90 @@ data "coder_git_auth" "github" {
   id = "github"
 }
 ```
+
+## GitHub clone with branch name
+
+To GitHub clone with a specific branch like `feat/example`
+
+```tf
+# Prompt the user for the git repo URL
+data "coder_parameter" "git_repo" {
+  name         = "git_repo"
+  display_name = "Git repository"
+  default      = "https://github.com/coder/coder/tree/feat/example"
```

Comment:
Should we add an optional `branch_name` in case the user wants to be explicit about it?  No harm in supporting both in the URL and a separate param, I think.

Reply:
True. I can add that in for cases where this was handled elsewhere